internal/exec/atlantis_generate_repo_config.go

```diff
@@ -0,0 +1,236 @@
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+// ExecuteAtlantisGenerateRepoConfigCmd executes `atlantis generate repo-config` command
+func ExecuteAtlantisGenerateRepoConfigCmd(cmd *cobra.Command, args []string) error {
+	flags := cmd.Flags()
+
+	configTemplateName, err := flags.GetString("config-template")
+	if err != nil {
+		return err
+	}
+
+	projectTemplateName, err := flags.GetString("project-template")
+	if err != nil {
+		return err
+	}
+
+	workflowTemplateName, err := flags.GetString("workflow-template")
+	if err != nil {
+		return err
+	}
+
+	stacksCsv, err := flags.GetString("stacks")
+	if err != nil {
+		return err
+	}
+	var stacks []string
+	if stacksCsv != "" {
+		stacks = strings.Split(stacksCsv, ",")
+	}
+
+	componentsCsv, err := flags.GetString("components")
+	if err != nil {
+		return err
+	}
+	var components []string
+	if componentsCsv != "" {
+		components = strings.Split(componentsCsv, ",")
+	}
+
+	return ExecuteAtlantisGenerateRepoConfig(configTemplateName, projectTemplateName, workflowTemplateName, stacks, components)
+}
+
+// ExecuteAtlantisGenerateRepoConfig generates repository configuration for Atlantis
+func ExecuteAtlantisGenerateRepoConfig(configTemplateName string, projectTemplateName string, workflowTemplateName string, stacks []string, components []string) error {
+	var configAndStacksInfo c.ConfigAndStacksInfo
+	stacksMap, err := FindStacksMap(configAndStacksInfo, false)
+	if err != nil {
+		return err
+	}
+
+	var configTemplate c.AtlantisRepoConfig
+	var projectTemplate c.AtlantisProjectConfig
+	var workflowTemplate any
+	var ok bool
+
+	if configTemplate, ok = c.Config.Integrations.Atlantis.ConfigTemplates[configTemplateName]; !ok {
+		return errors.Errorf("atlantis config template '%s' is not defined in 'integrations.atlantis.config_templates' in atmos.yaml", configTemplateName)
+	}
+
+	if projectTemplate, ok = c.Config.Integrations.Atlantis.ProjectTemplates[projectTemplateName]; !ok {
+		return errors.Errorf("atlantis project template '%s' is not defined in 'integrations.atlantis.project_templates' in atmos.yaml", projectTemplateName)
+	}
+
+	if workflowTemplate, ok = c.Config.Integrations.Atlantis.WorkflowTemplates[workflowTemplateName]; !ok {
+		return errors.Errorf("atlantis workflow template '%s' is not defined in 'integrations.atlantis.workflow_templates' in atmos.yaml", workflowTemplateName)
+	}
+
+	var atlantisProjects []c.AtlantisProjectConfig
+	atlantisWorkflows := map[string]any{}
+
+	// Iterate over all components in all stacks and generate atlantis projects
+	for stackConfigFileName, stackSection := range stacksMap {
+		if componentsSection, ok := stackSection.(map[any]any)["components"].(map[string]any); ok {
+			if terraformSection, ok := componentsSection["terraform"].(map[string]any); ok {
+				for componentName, compSection := range terraformSection {
+					if componentSection, ok := compSection.(map[string]any); ok {
```

Comment:
Since there are no `else` clauses, can we turn this nested if statement into a single if statement separated by &&s ? It's hard to read at the moment as the nest keeps left indenting the code.